libs/cart/src/drivers/magento/cart.service.spec.ts

```diff
@@ -0,0 +1,201 @@
+  let mockCartResponse: MagentoGetCartResponse;
+  let mockCreateCartResponse: MagentoCreateCartResponse;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        ApolloTestingModule
+      ],
+      providers: [
+        DaffMagentoCartService,
+        {
+          provide: DaffMagentoCartTransformer,
+          useValue: jasmine.createSpyObj('DaffMagentoCartTransformer', ['transform'])
+        },
+        {
+          provide: DaffCartItemDriver,
+          useValue: jasmine.createSpyObj('DaffCartItemDriver', ['delete', 'list'])
+        }
+      ]
+    });
+
+    service = TestBed.get(DaffMagentoCartService);
+    controller = TestBed.get(ApolloTestingController);
+
+    magentoCartTransformerSpy = TestBed.get(DaffMagentoCartTransformer);
+    magentoCartItemSpy = TestBed.get(DaffCartItemDriver);
+
+    daffCartFactory = TestBed.get(DaffCartFactory);
+    magentoCartFactory = TestBed.get(MagentoCartFactory);
+    magentoCartItemFactory = TestBed.get(MagentoCartItemFactory);
+    daffCartItemFactory = TestBed.get(DaffCartItemFactory);
+
+    mockDaffCart = daffCartFactory.create();
+    mockMagentoCart = magentoCartFactory.create();
+    mockMagentoCartItem = magentoCartItemFactory.create();
+    mockDaffCartItem = daffCartItemFactory.create();
+
+    mockMagentoCartItem.id = String(mockDaffCartItem.item_id);
+    cartId = mockDaffCart.id;
+    mockMagentoCart.items = [mockMagentoCartItem];
+    mockDaffCart.items = [mockDaffCartItem];
+    mockCreateCartResponse = {
+      createEmptyCart: String(cartId)
+    };
+    mockCartResponse = {
+      cart: mockMagentoCart
+    };
+
+    magentoCartTransformerSpy.transform.withArgs(mockMagentoCart).and.returnValue(mockDaffCart);
+    magentoCartItemSpy.list.withArgs(cartId).and.returnValue(of(mockDaffCart.items))
+    magentoCartItemSpy.delete.and.callFake((_, itemId) => of({
+      ...mockDaffCart,
+      items: mockDaffCart.items.filter(({item_id}) => item_id !== itemId)
+    }));
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  describe('get | getting a single cart', () => {
+    it('should call the transformer with the correct argument', done => {
+      service.get(cartId).subscribe(() => {
+        expect(magentoCartTransformerSpy.transform).toHaveBeenCalledWith(mockCartResponse.cart);
+        done();
+      });
+
+      const op = controller.expectOne(getCart);
+
+      op.flush({
+        data: mockCartResponse
+      });
+    });
+
+    it('should return the correct value', done => {
+      service.get(cartId).subscribe(result => {
+        expect(result).toEqual(jasmine.objectContaining(mockDaffCart));
+        done();
+      });
+
+      const op = controller.expectOne(getCart);
+
+      op.flush({
+        data: mockCartResponse
+      });
+    });
+
+    afterEach(() => {
+      controller.verify();
+    });
+  });
+
+  // skip deprecated method
+  xdescribe('addToCart | adding a product to the cart', () => {
```

Comment:
might as well delete this

Reply:
Changed in [`fc1be1c`](https://github.com/graycoreio/daffodil/pull/677/commits/fc1be1c3f1348ee04e614548f2b6db099c423535).